lib/enumerable.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+# This module recreates each, each_with_index, select,
+# all, any, none, count, map and inject enumerable methods.
+module Enumerable
+  def my_each
+    each = []
+    if block_given?
+      self.each do |i|
```

Comment:
Wh ydid you remove your original custom and replace it with this?

You are not supposed to use original enumerable method for your custom method.


I noticed before you had
```ruby
def my_each
    for i in self
      yield i
    end
  end
```

which is fine, you simply have to return self at the end and handle, if you so choose if a block is not given.


Because of this change I'd have to ask you to redo the tests for your actual custom method, not using each